### PR TITLE
Tests on Windows

### PR DESCRIPTION
**Pull request by justincredible**
I have updated glium [#2052](https://github.com/glium/glium/pull/2052) to get tests and doc tests running successfully on Windows just by running `cargo test`.

This branch replicates the approach used there based off your changes but probably needs some refining. The test `multiple_displays` passes in the above PR, but here it is still failing. I also noticed some intermittent test failures that I will look into. The changes in [#1](https://github.com/Melchizedek6809/glium/pull/1) are still relevant, but I did not include them in this branch.

Sorry I am unable to test this on a Unix system at this time, but I believe I have addressed the non-portable aspects.

### CODE REVIEW

**Comment by justincredible** [comment]
All tests are passing except the four `empty_stenciltexture*` tests.

**Comment by Melchizedek6809** [comment]
I'm still getting the following errors for most tests on my Win11 ARM64 machine:

```
---- attribute_vec4_tuple_i8 stdout ----
thread 'attribute_vec4_tuple_i8' panicked at 'internal error: entered unreachable code', /rustc/69f9c33d71c871fc16ac445211281c6e7a340943\library\std\src\sync\mpsc\oneshot.rs:190:22
```

The machine it self runs most example, any idea what could be the cause here? I've tried running them with `cargo test`.

**Comment by justincredible** [comment]
That std library file no longer exists since Nov. 9, 2022. You can update Rust, but I suspect it could still fail. The comment above the unreachable case is `We are the sole receiver; there cannot be a blocking receiver already.`, so I think `CONFIG_RECEIVER` should be wrapped in a `Mutex` instead of a `RwLock`.

**Comment by Melchizedek6809** [comment]
Nice, just updated my toolchain and all tests pass except the 4 previously mentioned ones

Are you already looking into what causes those 4 to fail? I'll take a look at getting these to run on Linux as well since you've already done all the hard work :)

**Comment by justincredible** [comment]
I have not looked very deeply into these macro tests, but I can try to resolve them over the weekend.

**Comment by Melchizedek6809** [comment]
Sorry for the force-push, rebased on the current glium:master since a couple of commits started accumulating there.

**Comment by justincredible** [comment]
No worries, all tests are working for me now. The proposed change to src/texture/get_format.rs seems okay.

I went ahead with using `Mutex` on the `Receiver` as it's still marked `!Sync`. Hopefully we'll see an `mpmc` exposed soon, as it looks to be implemented privately already.

File: src/texture/get_format.rs
Comment (Melchizedek6809) [inline src/texture/get_format.rs:1]: I'm wondering what these changes are for, did you encounter a bug here while fixing the test cases?

**Review by Melchizedek6809 — COMMENTED** [review]
Looks pretty good, just wondering what problem the changes in `src/texture/get_format.rs` are adressing?

File: src/texture/get_format.rs
Comment (justincredible) [reply src/texture/get_format.rs:1]: The four tests for `empty_stenciltexture1d` etc. were panicking on my system from calling `from_glenum` with `NONE`.

Comment (justincredible) [reply src/texture/get_format.rs:1]: I should have done my due diligence as this is an old known issue [glium #1240](https://github.com/glium/glium/issues/1240).

Perhaps a more appropriate fix for the panics would simply be:
`src/texture/get_format.rs`
``` rust
         Ok(match (red_sz, red_ty, green_sz, green_ty, blue_sz, blue_ty,
                alpha_sz, alpha_ty, depth_sz, depth_ty)
         {
+            (_, gl::NONE, _, _, _, _, _, _, _, gl::NONE) => return Err(GetFormatError::NotSupported),
             (_, gl::NONE, _, _, _, _, _, _, sz1, ty1) => InternalFormat::OneComponent {
                 ty1: InternalFormatType::from_glenum(ty1),
                 bits1: sz1 as usize,
```

**Comment by justincredible** [comment]
I'm moving this PR back to glium proper.

**Pull request closed without merging**